packages/flutter/test/material/flexible_space_bar_stretch_mode_test.dart

```diff
@@ -0,0 +1,172 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+final Key blockKey = UniqueKey();
+const double expandedAppbarHeight = 250.0;
+final Key finderKey = UniqueKey();
+
+void main() {
+  testWidgets('FlexibleSpaceBar stretch mode default zoomBackground', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: CustomScrollView(
+            physics: const BouncingScrollPhysics(),
+            key: blockKey,
+            slivers: <Widget>[
+              SliverAppBar(
+                expandedHeight: expandedAppbarHeight,
+                pinned: true,
+                stretch: true,
+                flexibleSpace: FlexibleSpaceBar(
+                  background: Container(
+                    key: finderKey,
+                  ),
+                ),
+              ),
+              SliverToBoxAdapter(child: Container(height: 10000.0)),
+            ],
+          ),
+        ),
+      ),
+    );
+
```

Comment:
Maybe add what's happening here like:

`// Scrolling up past the top causes the AppBar to shrink.`

Or put it in the title string? Up to you if you think it's helpful or not.